src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java

```diff
@@ -99,15 +105,7 @@ public void onFinalized(Run run) {
             if (log.getName().equals("log")) {
                 LOGGER.log(Level.FINE, String.format("Compressing build log of %s", run));
 
-                // regular, expected log file
-                FileInputStream fis = null;
-                FileOutputStream fos = null;
-                GZIPOutputStream gzos = null;
-
-                try {
-                    fis = new FileInputStream(log);
-                    fos = new FileOutputStream(new File(log.getParentFile(), gzippedLogName));
-                    gzos = new GZIPOutputStream(fos);
+                try(FileInputStream fis = new FileInputStream(log); GZIPOutputStream gzos = new GZIPOutputStream(new FileOutputStream(new File(log.getParentFile(), gzippedLogName)))) {
```

Comment:
```suggestion
                try(FileInputStream fis = new FileInputStream(log); FileOutputStream fos = new FileOutputStream(new File(log.getParentFile(), gzippedLogName)); GZIPOutputStream gzos = new GZIPOutputStream(fos)) {
```

to ensure `fos` gets closed